### PR TITLE
main: Use time.Duration for time-related flags

### PR DESCRIPTION
**Pull request by gvso**
This updates the type expected by the `cli-run-interval` and `healthcheck-offset` flags to time.Duration. This makes it more logical to the user and simplify our current implementation.

### CODE REVIEW

**Comment by ahmetb** [comment]
Still need to validate negative times which are negative inputs to our program :) 
https://play.golang.org/p/fkj1tmcd8fQ

**Comment by gvso** [comment]
`cli-run-interval` was never checked before, which is actually a bug. So, I agree we should check it now.

The health check offset is checked in `config.Validate`. Should we check it before?

**Comment by ahmetb** [comment]
Overall, would be definitely good to check these as early as possible (e.g. before a server starts etc)

**Comment by gvso** [comment]
The configuration is always validated right after the flags are validated. I added validation for both flags though.

File: README.md [hunk 1]
```diff
@@ -91,10 +91,10 @@ available Cloud Run regions](https://cloud.google.com/run/docs/locations))
...
 
-- `-cli-run-interval`: The time between each health check, in seconds (default:
-`60`). This is only need it if running with `-cli` option.
-- `-healthcheck-offset`: To evaluate the candidate's health, use metrics from
-the last `N` minutes relative to current rollout process (default: `30`)
+- `-cli-run-interval`: The time between each health check (default: `60s`). This
+is only need it if running with `-cli` option.
+- `-healthcheck-offset`:range of time for retrieving metrics relative to current
+rollout process (default: `30m`)
 - `-min-requests`: The minimum number of requests needed to determine the
```
Comment (ahmetb) [inline README.md:99]: probably make a note that 

> [...] needed (with -healthcheck-offset) to determine

also worth adding to cli help

File: README.md [hunk 2]
```diff
@@ -91,10 +91,10 @@ available Cloud Run regions](https://cloud.google.com/run/docs/locations))
 
 The rollout strategy consists of the steps and health criteria.
 
-- `-cli-run-interval`: The time between each health check, in seconds (default:
-`60`). This is only need it if running with `-cli` option.
-- `-healthcheck-offset`: To evaluate the candidate's health, use metrics from
-the last `N` minutes relative to current rollout process (default: `30`)
+- `-cli-run-interval`: The time between each health check (default: `60s`). This
+is only need it if running with `-cli` option.
+- `-healthcheck-offset`:range of time for retrieving metrics relative to current
```
Comment (ahmetb) [inline README.md]: I'd probably rewrite this as 

> Duration to look back for metrics to assess health of the candidate revision.

or

> time window to look back at the time of health check to assess health of the candidate revision.

**Comment by ahmetb** [comment]
FYI  56a973f commit description doesn't match the diff

File: README.md (hunk 2 above)
Comment (gvso) [reply README.md]: I like second more but would rephrase it to

> time window to look back during health check to assess the candidate revision's health.

to avoid the repetition of the word "time" in different contexts and to separate the occurrences of "health" a little bit more.

File: README.md (hunk 1 above)
Comment (gvso) [reply README.md:195]: What exactly do you mean by this? What's needed with `healthcheck-offset`?

Comment (ahmetb) [reply README.md:195]: I meant that it is worth noting that min-requests are calculated "within $healthcheck-offset minutes" and not all-time.

Comment (gvso) [reply README.md:195]: Oh sure. Let me add that piece
